exercises/dominoes/test/Tests.hs

```diff
@@ -0,0 +1,108 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE RecordWildCards #-}
+
+import Control.Monad     (forM_, unless)
+import Data.Foldable     (for_)
+import Data.Function     (on)
+import Test.Hspec        (Spec, describe, expectationFailure, it, shouldBe, shouldMatchList)
+import Test.Hspec.Runner (configFastFail, defaultConfig, hspecWith)
+import Text.Printf       (printf)
+
+import Dominoes (chain)
+
+main :: IO ()
+main = hspecWith defaultConfig {configFastFail = True} specs
+
+specs :: Spec
+specs = describe "dominoes" $
+          describe "chain" $ for_ cases test
+  where
+
+    test Case{..} = it description assertion
+      where
+        assertion = if expected
+                      then shouldBeChain $ chain input
+                      else chain input `shouldBe` Nothing
```

Comment:
Nice solution to the problem of matching `Maybe [a]` disregarding the order BTW. :+1: